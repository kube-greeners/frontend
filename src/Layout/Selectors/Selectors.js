import React from 'react'
import { Col, Row, Select } from 'antd'
import { useSelector } from 'react-redux'
const { Option } = Select;

export default function Selectors() {

    const {namespaces,resources} = useSelector(state => state.dashboard.selects);
    return (
        <>
            <Row gutter={24} style={{ paddingTop: '7rem' }}>
                <Col span={7}>
                    <Selector data={namespaces.data} name="Namespace" defaultVal={namespaces.currentlySelected} style={{display:'block'}}/>
                </Col>
                <Col span={7}>
                    <Selector data={resources.data} name="Resource" defaultVal={resources.currentlySelected} style={{display:'block'}}/>
                </Col>
            </Row>
        </>
    )
}

function Selector({ data, name,defaultVal , ...rest }) {

    const labelStyle = {
        marginBottom:'.5rem',
        display:'block',
        color:'#666666'
    }

    return (
        <>
            <label for={name} style={labelStyle}>{name}</label>
            <Select id={name} defaultValue={defaultVal} {...rest} >
                {data.map(n => <Option value={n}>{n}</Option>)}
            </Select>
        </>
    )
}
