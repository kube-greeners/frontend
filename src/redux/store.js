import { configureStore } from '@reduxjs/toolkit'
import dashboardReducer from './dashboardSlice'
import { apiSlice } from './apiSlice'

export default configureStore({
  reducer: {
    dashboard: dashboardReducer,
    [apiSlice.reducerPath]:apiSlice.reducer
  },
  middleware: getDefaultMiddleware =>
  getDefaultMiddleware().concat(apiSlice.middleware)
})