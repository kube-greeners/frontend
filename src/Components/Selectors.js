import React from 'react'
import { Col, Row, Select, DatePicker } from 'antd'
import { useDispatch, useSelector } from 'react-redux'
import { setCurrentlySelectedNamespace, setCurrentlySelectedResource, setCurrentInterval } from '../redux/dashboardSlice';
import moment from 'moment';
import { convertDateWithOutTimestamp } from '../Utilities/utilityFunctions';
const { Option } = Select;

export default function Selectors() {

    const {RangePicker} = DatePicker;

    const dateFormat = 'DD/MM/YYYY';

    const {namespaces,resources} = useSelector(state => state.dashboard.selects);
    const {startDateUnix, endDateUnix} = useSelector(state => state.dashboard.interval);
    const startDate = convertDateWithOutTimestamp(startDateUnix)
    const endDate = convertDateWithOutTimestamp(endDateUnix)


    const dispatch = useDispatch();
    
    const nameSpaceSelected = ns => {
        dispatch(setCurrentlySelectedNamespace(ns))
    }

    const resourceSelected = rs => {
        dispatch(setCurrentlySelectedResource(rs))
    }

    const intervalSelected = (date, dateString) => {
        dispatch(setCurrentInterval(dateString))

        
    }
    function disabledDate(current) {
        return current && current > moment().endOf('day');
      }

    const labelStyle = {
        marginBottom:'.5rem',
        display:'block',
        color:'#666666'
    }

    return (
        <>
            <Row gutter={24} style={{ paddingTop: '7rem' }}>
                <Col span={7}>
                    <Selector 
                    data={namespaces.data} 
                    name="Namespace" 
                    defaultVal={namespaces.currentlySelected} 
                    labelStyle={labelStyle}
                    style={{ display: 'block' }} 
                    onChange={nameSpaceSelected} />
                </Col>
                <Col span={7}>
                    <Selector
                        data={resources.data}
                        name="Resource" 
                        defaultVal={resources.currentlySelected} 
                        labelStyle={labelStyle}
                        style={{ display: 'block' }} 
                        onChange={resourceSelected} />
                </Col>
                <Col span={7}>
                    <label style={labelStyle}>Time interval</label>
                    <RangePicker
                        defaultValue={[moment(startDate, dateFormat), moment(endDate, dateFormat)]}
                        format={dateFormat}
                        disabledDate={disabledDate}
                        onChange={intervalSelected}/>    
                </Col>
            </Row>
        </>
    )
}

function Selector({ data, name, defaultVal, labelStyle, ...rest }) {

    return (
        <>
            <label htmlFor={name} style={labelStyle}>{name}</label>
            <Select id={name} defaultValue={defaultVal} {...rest} >
                {data.map(n => <Option key={n} value={n}>{n}</Option>)}
            </Select>
        </>
    )
}
