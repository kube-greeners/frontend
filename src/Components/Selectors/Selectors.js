import React from "react";
import { Col, Row, Select, DatePicker } from "antd";
import { useDispatch, useSelector } from "react-redux";
import {
  setCurrentlySelectedNamespace,
  setCurrentInterval,
} from "../../redux/dashboardSlice";
import { useGetNameSpacesQuery } from "../../redux/apiSlice";
import moment from "moment";
import { convertDateWithoutTimestamp } from "../../Utilities/utilityFunctions";
const { Option } = Select;

export default function Selectors() {
  const { RangePicker } = DatePicker;

  const dateFormat = "DD/MM/YYYY";

  const { namespaces } = useSelector((state) => state.dashboard.selects);
  const { startDateUnix, endDateUnix } = useSelector(
    (state) => state.dashboard.interval
  );

  const startDate = convertDateWithoutTimestamp(startDateUnix);
  const endDate = convertDateWithoutTimestamp(endDateUnix);

  const namespaceNamesFetch = useGetNameSpacesQuery({
    startDate: startDateUnix,
    endDate: endDateUnix,
  });

  const dispatch = useDispatch();

  const nameSpaceSelected = (ns) => {
    dispatch(setCurrentlySelectedNamespace(ns));
  };

  const intervalSelected = (date, dateString) => {
    dispatch(setCurrentInterval(dateString));
  };

  function disabledDate(current) {
    return current > moment().endOf("day");
  }

  const labelStyle = {
    marginBottom: ".5rem",
    display: "block",
    color: "#666666",
  };

  return (
    <>
      <Row gutter={24} style={{ paddingTop: "7rem" }}>
        <Col span={7}>
          <Selector
            data={namespaceNamesFetch.data}
            name="Namespace"
            loading={namespaceNamesFetch.isFetching}
            defaultVal={namespaces.currentlySelected}
            labelStyle={labelStyle}
            style={{ display: "block" }}
            onChange={nameSpaceSelected}
          />
        </Col>
        <Col span={7}>
          <label style={labelStyle}>Time interval</label>
          <RangePicker
            defaultValue={[
              moment(startDate, dateFormat),
              moment(endDate, dateFormat),
            ]}
            format={dateFormat}
            disabledDate={disabledDate}
            onChange={intervalSelected}
          />
        </Col>
      </Row>
    </>
  );
}

export function Selector({
  title=true,
  data,
  name,
  defaultVal,
  labelStyle,
  ...rest
}) {
  return (
    <>
      {title && (
        <label htmlFor={name} style={labelStyle}>
          {name}
        </label>
      )}
      <Select id={name} defaultValue={defaultVal} {...rest}>
        {data &&
          data.map((n) => (
            <Option key={n} value={n}>
              {n}
            </Option>
          ))}
      </Select>
    </>
  );
}