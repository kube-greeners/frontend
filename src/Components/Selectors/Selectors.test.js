import Selectors from "./Selectors";
import renderer from "react-test-renderer";
import store from "../../redux/store";
import { Provider } from "react-redux";

test("matches Selectors snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Selectors />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
