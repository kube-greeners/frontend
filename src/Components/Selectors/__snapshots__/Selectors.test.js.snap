// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Selectors snapshot 1`] = `
<div
  className="ant-row"
  style={
    Object {
      "marginLeft": -12,
      "marginRight": -12,
      "paddingTop": "7rem",
    }
  }
>
  <div
    className="ant-col ant-col-7"
    style={
      Object {
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
  >
    <label
      htmlFor="Namespace"
      style={
        Object {
          "color": "#666666",
          "display": "block",
          "marginBottom": ".5rem",
        }
      }
    >
      Namespace
    </label>
    <div
      className="ant-select ant-select-single ant-select-show-arrow"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="ant-select-selector"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-search"
        >
          <input
            aria-activedescendant="Namespace_list_0"
            aria-autocomplete="list"
            aria-controls="Namespace_list"
            aria-haspopup="listbox"
            aria-owns="Namespace_list"
            autoComplete="off"
            className="ant-select-selection-search-input"
            id="Namespace"
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={true}
            role="combobox"
            style={
              Object {
                "opacity": 0,
              }
            }
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          className="ant-select-selection-item"
          title="All namespaces"
        >
          All namespaces
        </span>
      </div>
      <span
        aria-hidden={true}
        className="ant-select-arrow"
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <span
          aria-label="down"
          className="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
  <div
    className="ant-col ant-col-7"
    style={
      Object {
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
  >
    <label
      htmlFor="Resource"
      style={
        Object {
          "color": "#666666",
          "display": "block",
          "marginBottom": ".5rem",
        }
      }
    >
      Resource
    </label>
    <div
      className="ant-select ant-select-single ant-select-show-arrow"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="ant-select-selector"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-search"
        >
          <input
            aria-activedescendant="Resource_list_0"
            aria-autocomplete="list"
            aria-controls="Resource_list"
            aria-haspopup="listbox"
            aria-owns="Resource_list"
            autoComplete="off"
            className="ant-select-selection-search-input"
            id="Resource"
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={true}
            role="combobox"
            style={
              Object {
                "opacity": 0,
              }
            }
            type="search"
            unselectable="on"
            value=""
          />
        </span>
        <span
          className="ant-select-selection-item"
          title="Estimated CO2 emission"
        >
          Estimated CO2 emission
        </span>
      </div>
      <span
        aria-hidden={true}
        className="ant-select-arrow"
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <span
          aria-label="down"
          className="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
  <div
    className="ant-col ant-col-7"
    style={
      Object {
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
  >
    <label
      style={
        Object {
          "color": "#666666",
          "display": "block",
          "marginBottom": ".5rem",
        }
      }
    >
      Time interval
    </label>
    <div
      className="ant-picker ant-picker-range"
      onClick={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ant-picker-input ant-picker-input-active"
      >
        <input
          autoComplete="off"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          placeholder="Start date"
          readOnly={true}
          size={12}
          value=""
        />
      </div>
      <div
        className="ant-picker-range-separator"
      >
        <span
          aria-label="to"
          className="ant-picker-separator"
        >
          <span
            aria-label="swap-right"
            className="anticon anticon-swap-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="swap-right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        className="ant-picker-input"
      >
        <input
          autoComplete="off"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          placeholder="End date"
          readOnly={true}
          size={12}
          value=""
        />
      </div>
      <div
        className="ant-picker-active-bar"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <span
        className="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          className="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
      <span
        className="ant-picker-clear"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <span
          aria-label="close-circle"
          className="anticon anticon-close-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;
